tests/translate_ciphers.py

```diff
@@ -0,0 +1,119 @@
+# limitations under the License.
+#
+# Purpose
+#
+# Translate ciphersuite names in MBedTLS format to OpenSSL and GNU
+# standards.
+#
+# Format and analyse strings past in via input arguments to match
+# the expected strings utilised in compat.sh.
+#
+# sys.argv[1] should be "g" or "o" for GNU or OpenSSL.
+# sys.argv[2] should be a string containing one or more
+# ciphersuite names.
+
+import re
+import sys
+
+def translate_gnu(m_cipher):
+    # Remove "TLS-"
+    # Replace "-WITH-" with ":+"
+    # Remove "EDE"
+    m_cipher = "+" + m_cipher[4:]
+    m_cipher = m_cipher.replace("-WITH-", ":+")
+    m_cipher = m_cipher.replace("-EDE", "")
+
+    # SHA == SHA1, if the last 3 chars are SHA append 1
+    if m_cipher[-3:] == "SHA":
+        m_cipher = m_cipher+"1"
+
+    # CCM or CCM-8 should be followed by ":+AEAD"
+    if "CCM" in m_cipher:
+        m_cipher = m_cipher+":+AEAD"
+
+    # Replace the last "-" with ":+"
+    # Replace "GCM:+SHAxyz" with "GCM:+AEAD"
+    else:
+        index=m_cipher.rindex("-")
+        m_cipher = m_cipher[:index]+":+"+m_cipher[index+1:]
+        m_cipher = re.sub(r"GCM\:\+SHA\d\d\d", "GCM:+AEAD", m_cipher)
+
+    return m_cipher
+
+def translate_ossl(m_cipher):
+    # Remove "TLS-"
+    # Remove "WITH"
+    m_cipher = m_cipher[4:]
+    m_cipher = m_cipher.replace("-WITH", "")
+
+    # Remove the "-" from "ABC-xyz"
+    m_cipher = m_cipher.replace("AES-", "AES")
+    m_cipher = m_cipher.replace("CAMELLIA-", "CAMELLIA")
+    m_cipher = m_cipher.replace("ARIA-", "ARIA")
+
+    # Remove "RSA" if it is at the beginning
+    if m_cipher[:4] == "RSA-":
+        m_cipher = m_cipher[4:]
+
+    # For all circumstances outside of PSK
+    if "PSK" not in m_cipher:
+        m_cipher = m_cipher.replace("-EDE", "")
+        m_cipher = m_cipher.replace("3DES-CBC", "DES-CBC3")
+
+        # Remove "CBC" if it is not prefixed by DES
+        if "CBC" in m_cipher:
+            index = m_cipher.rindex("CBC")
+            if m_cipher[index-4:index-1] != "DES":
+                m_cipher = m_cipher.replace("CBC-", "")
+
+    # ECDHE-RSA-ARIA does not exist in OpenSSL
+    m_cipher = m_cipher.replace("ECDHE-RSA-ARIA", "ECDHE-ARIA")
+
+    # POLY1305 should not be followed by anything
+    if "POLY1305" in m_cipher:
+        index = m_cipher.rindex("POLY1305")
+        m_cipher=m_cipher[:index+8]
+
+    # If DES is being used, Replace DHE with EDH
+    if "DES" in m_cipher and "DHE" in m_cipher and "ECDHE" not in m_cipher:
+        m_cipher = m_cipher.replace("DHE", "EDH")
+
+    return m_cipher
+
+def format(mode, ciphers):
+    ciphers = ciphers.split()
+    t_ciphers = []
+    if mode == "g":
+        for i in ciphers:
+            t_ciphers.append(translate_gnu(i))
+    if mode == "o":
+        for i in ciphers:
+            t_ciphers.append(translate_ossl(i))
+    return " ".join(t_ciphers)
+
+def main():
+    # print command line arguments
+    if len(sys.argv) <= 2:
+        exit(1)
```

Comment:
Done, but I still need to modify the bash script to error if it receives an exit 1 state

Reply:
I've added a check in the bash script on my end, but is it worth making the check in `compat.sh`? It is not like it relies in user input. As long as it is coded correctly it will function correctly.

Reply:
I had forgotten about that, thanks for raising the question. I think it would be safer to check the exit status in compat.sh indeed. Otherwise, if we ever end up making an incorrect change, it could result in the list of Gnu TLS and OpenSSL ciphers being empty, which would be very easy to miss (we rarely run `compat.sh` manually, and the CI wouldn't complain about the number of test cases run decreasing suddenly).